File: api/src/main/java/com/grash/model/ApiKey.java
package com.grash.model;

import com.grash.model.abstracts.CompanyAudit;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@NoArgsConstructor
public class ApiKey extends CompanyAudit {
    @NotNull
    private String label;
    @NotNull
    private String code;
    @ManyToOne
    @NotNull
    private OwnUser user;
    private Date lastUsed;
}

File: frontend/src/models/owns/apiKey.ts
import { Audit } from './audit';
import { UserMiniDTO } from './user';

export interface ApiKey extends Audit {
  id: number;
  label: string;
  code: string;
  user: UserMiniDTO;
  lastUsed?: Date;
  expiresAt: string;
  revokedAt: string;
}

export interface ApiKeyPostDTO {
  label: string;
  expiresAt?: string;
}

export interface ApiKeyShowDTO extends Audit {
  id: number;
  label: string;
  code?: string;
  user: UserMiniDTO;
  lastUsed?: Date;
  expiresAt?: string;
  revokedAt?: string;
}

export interface ApiKeyCriteria {
  active?: boolean;
}
